src/components/Swap/Toolbar/Toolbar.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { TradeType } from '@uniswap/sdk-core'
+import { DAI, USDC_MAINNET } from 'constants/tokens'
+import { SwapInfoProvider } from 'hooks/swap/useSwapInfo'
+import { flagsAtom } from 'hooks/useSyncFlags'
+import Module from 'module'
+import { Field, stateAtom, Swap } from 'state/swap'
+import { renderComponent } from 'test'
+
+import Toolbar from './index'
+
+jest.mock('@web3-react/core', () => ({
+  ...(jest.requireActual('@web3-react/core') as Module),
+  useWeb3React: () => ({
+    isActive: true,
+    chainId: 1,
+    account: '0x0000000000000000000000000000000000000000',
+  }),
+}))
+
+jest.mock('../../../hooks/usePermit2Allowance', () => ({
+  __esModule: true,
+  ...(jest.requireActual('../../../hooks/usePermit2Allowance') as Module),
+  default: () => ({
+    state: 1 /* AllowanceState.REQUIRED */,
```

Comment:
Can you use the actual value? It would be much less brittle. I don't know if this is possible without moving the enum to a separate file.

```
    state: requireActual('../../../hooks/usePermit2Allowance').AllowanceState.REQUIRED
```